tests/test_tasks/test_task_functions.py

```diff
@@ -234,3 +234,9 @@ def test_deletion_of_cache_dir(self):
         self.assertTrue(os.path.exists(tid_cache_dir))
         openml.utils._remove_cache_dir_for_id("tasks", tid_cache_dir)
         self.assertFalse(os.path.exists(tid_cache_dir))
+
+    def test_supervised_task_target_format(self):
```

Comment:
Why is this a test for a task and not for the respective dataset?

Reply:
Also, do we have sufficient tests that the data is loaded correctly for sparse datasets?

Reply:
I added this unit test to datasets: https://github.com/openml/openml-python/blob/fix_1057/tests/test_datasets/test_dataset.py#L306

and removed this unit test from tasks as this new unit test makes the one under tasks redundant
I also checked this unit test with the dataset used in the #1057 and this unit test passes for it too as it is a more generalized test given the changes made in this PR